core/src/main/kotlin/fr/sncf/osrd/stdcm/infra_exploration/InfraExplorerWithEnvelopeImpl.kt

```diff
@@ -52,11 +53,25 @@ data class InfraExplorerWithEnvelopeImpl(
         val cached = envelopeCache?.get()
         if (cached != null) return cached
         val res = EnvelopeConcat.fromLocated(envelopes.toList())
-        val withStops = EnvelopeStopWrapper(res, stops)
+        val withStops = EnvelopeStopWrapper(res, generateTrainStops())
         envelopeCache = SoftReference(withStops)
         return withStops
     }
 
+    private fun generateTrainStops(): List<TrainStop> {
+        val steps = getStepTracker().getAllReachedSteps()
+        val stopOffsets = steps.filter { it.originalStep.stop }.map { it.pathOffset }
+        val stopDurations = stopTimeData
```

Comment:
From the name, I'd expect
```suggestion
        val stopDurations = stopTimeData.map {it.currentDuration}
```
(or zip directly on `stopTimeData.currentDuration` and get rid of this line)